walker.go

```diff
@@ -0,0 +1,57 @@
+package chi
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+type WalkFn func(method string, route string, handler http.Handler, middlewares ...func(http.Handler) http.Handler) error
+
+func Walk(r Routes, fn WalkFn) error {
```

Comment:
`Walk()` gives you:
- full expanded flat routes with their middleware stack (with `chi.Routes`, developer would have to traverse all the nested routes himself and build the full path and middleware stack via recursion/loop -- which would be duplicated work)
- is very convenient for 1. custom docgen generation; 2. writing automatic tests (eg. testing all `/admin/users` endpoints and making sure that there's a certain middleware applied)
- gives more power to developers, since they can customize docgen generation process easily... for example they can filter out routes that are not public, remove TODOs from comments etc. on the fly -- they can even create two different docs: 1. Internal Developer Docs (with details about Handlers, Middlewares and file structure) and 2. Public API Docs (with less info for the API consumers)
- the signature is very similar to https://golang.org/pkg/path/filepath/#Walk

btw: I used to have Walk() func in a separate walker pkg in the beginning. But it makes a lot of sense to me that it should be part of the core.